controllers/openstackmachine_controller.go

```diff
@@ -328,9 +336,24 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{}, err
 	}
 
-	instanceStatus, err := r.getOrCreate(scope.Logger(), cluster, openStackCluster, machine, openStackMachine, computeService, userData)
+	managedSecurityGroups := getManagedSecurityGroups(openStackCluster, machine, openStackMachine)
+	instanceTags := getInstanceTags(openStackMachine, openStackCluster)
+	portsStatus, err := r.getOrCreatePorts(scope.Logger(), clusterName, openStackCluster, openStackMachine.Spec.Ports, openStackMachine.Spec.Trunk, managedSecurityGroups, instanceTags, openStackMachine.Name, openStackMachine, networkingService)
 	if err != nil {
-		// Conditions set in getOrCreate
+		// Conditions set in getOrCreatePorts
+		return ctrl.Result{}, err
+	}
+	// TODO(emilien) do we want to deal with Ports Status like we do for the nova instance? Might be expensive...
+	// In the meantime, we consider that ports are ready if they are created.
+	conditions.MarkTrue(openStackMachine, infrav1.PortsReadyCondition)
```

Comment:
Should we check port status before passing the list to the `getOrCreateInstance`? Or are we good trusting that ports exist. Note that there is no port reconcile for now so if a user deletes the port manually, the port will never be re-created and it'll still be in `PortsStatus`. Feedback is welcome on that.

Reply:
Looks like `getOrCreatePorts()` will recreate ports if they were deleted. I'd say we're fine here.

Reply:
I might be missing something, but If the Ports were deleted and remained on `openStackMachine.Status.PortsStatus`  wouldn't  `getOrCreatePorts()` skip the creation?
If that is the case, maybe the content of PortStatus could be compared to the lists of Port retrieved by `device_id`?

Reply:
if the ports were removed (let's say by the user by mistake), even if the ports are re-created by the machine controller, we need to attach them to the instance (hotplug) or recreate the instance again with the new ports.
So you're right, right now the controller doesn't reconcile the ports if they are manually removed outside of CAPO since they would still be in status.

Yes, we could use the content of PortStatus to be compared to the lists of Port retrieved by `device_id`, although that's again a new neutron API operation. I'm fine adding it, but I'll wait for other reviewer's opinion on this one.